scripts/generate/templates.ts

```diff
@@ -0,0 +1,267 @@
+ */`;
+
+const genericMethodNames: Record<ClientFunctionVerbs, string> = {
+  get: 'getResource',
+  list: 'getListResource',
+  delete: 'deleteResource',
+  create: 'createResource',
+  update: 'updateResource',
+};
+
+function pascalCaseToCamelCase(str: string) {
+  const firstLowerCaseIndex = str
+    .split('')
+    .findIndex((char) => char === char.toLocaleLowerCase());
+
+  if (firstLowerCaseIndex === -1) return str;
+
+  const firstUpperCaseIndex = Math.max(firstLowerCaseIndex - 1, 1);
+
+  return (
+    str.substring(0, firstUpperCaseIndex).toLocaleLowerCase() +
+    str.substring(firstUpperCaseIndex)
+  );
+}
+
+export function formatGeneratedFileExport(fileName: string) {
+  return `export * from './${fileName}';\n`;
+}
+
+export function formatInterfaceName(resourceName: string): string {
+  return `I${resourceName[0].toLocaleUpperCase()}${resourceName.slice(1)}`;
+}
+
+export function formatTypesFileHeader(apiVersion: string): string {
+  return `${autoGeneratedWarningMessage}
+  
+  import * as metav1 from 'model/services/mapi/metav1';
+  
+  export const ApiVersion = '${apiVersion}';\n`;
+}
+
+export function formatResourceKindExport(resourceName: string) {
+  return `export const ${resourceName} = '${resourceName}';\n`;
+}
+
+export function formatListResourceExport(
+  resourceNames: IResourceNames
+): string {
+  const resourceInterfaceName = formatInterfaceName(resourceNames.kind);
+  const listResourceInterfaceName = formatInterfaceName(resourceNames.listKind);
+
+  return `${formatResourceKindExport(resourceNames.listKind)}
+export interface ${listResourceInterfaceName} extends metav1.IList<${resourceInterfaceName}> {
+  apiVersion: typeof ApiVersion;
+  kind: typeof ${resourceNames.listKind};
+}\n`;
+}
+
+export function getClientFunctionMethodName(
+  resourceName: string,
+  verb: ClientFunctionVerbs
+) {
+  switch (verb) {
+    case 'list':
+      return `get${resourceName}`;
+    default:
+      return `${verb}${resourceName}`;
+  }
+}
+
+export function formatClientFunctionFileHeader(
+  resourceName: string,
+  verb: ClientFunctionVerbs
+): string {
+  const genericMethodName = genericMethodNames[verb];
+  return `${autoGeneratedWarningMessage}
+
+import { IHttpClient } from 'model/clients/HttpClient';
+import { ${genericMethodName} } from 'model/services/mapi/generic/${genericMethodName}';
+import * as k8sUrl from 'model/services/mapi/k8sUrl';
+import { IOAuth2Provider } from 'utils/OAuth2/OAuth2';
+
+import { ${formatInterfaceName(resourceName)} } from '.';\n`;
+}
+
+export function formatClientFunctionGetMethod(
+  apiVersion: string,
+  resourceName: string,
+  resourceNamePlural: string,
+  namespaced: boolean,
+  verb: ClientFunctionVerbs
+): string {
+  const clientFunctionName = getClientFunctionMethodName(resourceName, verb);
+
+  return `export function ${clientFunctionName}(
+  client: IHttpClient,
+  auth: IOAuth2Provider,${namespaced ? '\nnamespace: string,' : ''}
```

Comment:
It would be nice to move `namespace` parameter to a separate line and remove `\n` formatting, to improve readability, e.g.:
```
  auth: IOAuth2Provider,
  ${namespaced ? 'namespace: string,' : ''}
  name: string
```